### PR TITLE
[DEBHAS-358] do not return error if no component detected

### PR DESCRIPTION
**Pull request by yangcao77**
### What does this PR do?:

Do not return error if no component detected in cdq. return cdq completed with a success status, with no component detected. 

```
Name:         componentdetectionquery-sample2
Namespace:    application-service-system
Labels:       <none>
Annotations:  <none>
API Version:  appstudio.redhat.com/v1alpha1
Kind:         ComponentDetectionQuery
Metadata:
  Creation Timestamp:  2023-05-17T13:57:33Z
  Generation:          1
  Managed Fields:
    API Version:  appstudio.redhat.com/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:git:
          .:
          f:url:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2023-05-17T13:57:33Z
    API Version:  appstudio.redhat.com/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:conditions:
    Manager:         manager
    Operation:       Update
    Subresource:     status
    Time:            2023-05-17T13:57:37Z
  Resource Version:  38124
  UID:               5918f016-94dc-4736-86dd-5a846321d993
Spec:
  Git:
    URL:  https://github.com/christianvogt/empty
Status:
  Conditions:
    Last Transition Time:  2023-05-17T13:57:33Z
    Message:               ComponentDetectionQuery is processing
    Reason:                Success
    Status:                True
    Type:                  Processing
    Last Transition Time:  2023-05-17T13:57:37Z
    Message:               ComponentDetectionQuery has successfully finished
    Reason:                OK
    Status:                True
    Type:                  Completed
Events:                    <none>
```

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/DEVHAS-358

### PR acceptance criteria:

- [ ] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

**Review by johnmcollier — COMMENTED** [review]
Rather than just completing quietly, I wonder if we should return an separate status message saying that no components were found?

File: controllers/componentdetectionquery_controller.go [hunk 1]
```diff
@@ -338,12 +338,14 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
...
 			}
 		}
-
-		err = r.updateComponentStub(req, &componentDetectionQuery, devfilesMap, devfilesURLMap, dockerfileContextMap, componentPortsMap)
-		if err != nil {
-			log.Error(err, fmt.Sprintf("Unable to update the component stub %v", req.NamespacedName))
-			r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, copiedCDQ, err)
-			return ctrl.Result{}, nil
+		// only update the componentStub when a component has been detected
+		if len(devfilesMap) != 0 || len(devfilesURLMap) != 0 || len(dockerfileContextMap) != 0 {
```
Comment (maysunfaisal) [inline controllers/componentdetectionquery_controller.go:342]: not really needed IMO because we loop over these map in the func but nevertheless its ok

**Comment by yangcao77** [comment]
@maysunfaisal @johnmcollier  Updated  the CDQ to return a more concrete status, may I get another review?

**Review by maysunfaisal — APPROVED** [review]

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**